osmomath/decimal.go

```diff
@@ -552,23 +560,49 @@ func (d BigDec) MustFloat64() float64 {
 // Dec returns the osmomath.Dec representation of a BigDec.
 // Values in any additional decimal places are truncated.
 func (d BigDec) Dec() Dec {
-	precisionDiff := PrecisionBigDec - PrecisionDec
-	precisionFactor := new(big.Int).Exp(big.NewInt(10), big.NewInt(int64(precisionDiff)), nil)
-
-	if precisionDiff < 0 {
-		panic("invalid decimal precision")
+	return d.DecWithPrecision(PrecisionDec)
```

Comment:
I feel like we should change to DecPrecision at some point 🤔

Reply:
hmm, but why though?

Reply:
Haha just my personal feeling, PrecisionDec with Dec as suffix makes it feel like it's a Dec object while DecPricision with the precision suffix makes it feel like its a Precision

Reply:
We have `PrecisionBigDec` as well that will need changes. I think we should discuss this in a separate issue / PR to avoid blocking this

Reply:
yeah, defintiely dont want to include it in this PR, but hopefully at some point we can have an extended discussion  on this

Reply:
https://github.com/osmosis-labs/osmosis/issues/6466